### ISSUE TITLE
panics with index out of range for ingresses with undefined spec.tls.hosts

### ISSUE THREAD

**Issue by majodev** [stakater/IngressMonitorController#200]
Hi,

looks like kubernetes supports `tls` definitions with undefined/empty `hosts` in their ingress objects. These definitions lead to an `index out of range` panic in IngressMonitorController.

StackTrace:

```bash
E0827 10:45:42.248172       1 runtime.go:66] Observed a panic: "index out of range" (runtime error: index out of range)
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:28
/go/src/github.com/stakater/IngressMonitorController/pkg/kube/wrappers/ingress-wrapper.go:31
/go/src/github.com/stakater/IngressMonitorController/pkg/kube/wrappers/ingress-wrapper.go:78
/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:155
/go/src/github.com/stakater/IngressMonitorController/pkg/controller/actions.go:42
<autogenerated>:1
/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:121
/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:105
/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:97
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 11 [running]:
github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x10174e0, 0x1932750)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/stakater/IngressMonitorController/pkg/kube/wrappers.(*IngressWrapper).tryGetTLSHost(...)
	/go/src/github.com/stakater/IngressMonitorController/pkg/kube/wrappers/ingress-wrapper.go:31
github.com/stakater/IngressMonitorController/pkg/kube/wrappers.(*IngressWrapper).GetURL(0xc42060bca0, 0x11bef80, 0x11bef90)
	/go/src/github.com/stakater/IngressMonitorController/pkg/kube/wrappers/ingress-wrapper.go:78 +0x74e
github.com/stakater/IngressMonitorController/pkg/controller.(*MonitorController).getMonitorURL(0xc420236000, 0x1131200, 0xc420522140, 0x0, 0xc420236000)
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:155 +0x149
github.com/stakater/IngressMonitorController/pkg/controller.ResourceUpdatedAction.handle(0x1131200, 0xc420522140, 0x0, 0x0, 0xc420236000, 0xc42060be10, 0xc420163700)
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/actions.go:42 +0xe3
github.com/stakater/IngressMonitorController/pkg/controller.(*ResourceUpdatedAction).handle(0xc420163840, 0xc420236000, 0xc420163840, 0x7f29466b6c08)
	<autogenerated>:1 +0x61
github.com/stakater/IngressMonitorController/pkg/controller.(*MonitorController).processNextItem(0xc420236000, 0x85b000)
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:121 +0xdb
github.com/stakater/IngressMonitorController/pkg/controller.(*MonitorController).runWorker(0xc420236000)
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:105 +0x2b
github.com/stakater/IngressMonitorController/pkg/controller.(*MonitorController).(github.com/stakater/IngressMonitorController/pkg/controller.runWorker)-fm()
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:97 +0x2a
github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4202ee000)
	/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4202ee000, 0x3b9aca00, 0x0, 0xa7b202020202001, 0xc420224120)
	/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4202ee000, 0x3b9aca00, 0xc420224120)
	/go/src/github.com/stakater/IngressMonitorController/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/stakater/IngressMonitorController/pkg/controller.(*MonitorController).Run
	/go/src/github.com/stakater/IngressMonitorController/pkg/controller/controller.go:97 +0x1dc
```

Steps to reproduce:

Create an ingress object with an empty/uncommented spec.tls.hosts:

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  [...]
spec:
  [...]
  tls:
  - secretName: tls-secret
    hosts: 
     # - test.com
```

[IngressTLS v1beta1 extensions](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#ingresstls-v1beta1-extensions)

**Comment by waseem-h** [comment]
Thanks for pointing out the bug @majodev . Feel free to create a PR to fix this or I will see if I can fix that when I have time :)

**Comment by usamaahmadkhan** [comment]
@majodev fixed in version 1.0.86. Kindly close this issue.

**Comment by majodev** [comment]
awesome, thx!
